addons/pos_self_order/static/src/kiosk/pages/table_number/table_number.js

```diff
@@ -0,0 +1,44 @@
+/** @odoo-module */
+
+import { Component, useState } from "@odoo/owl";
+import { useselfOrder } from "@pos_self_order/kiosk/self_order_kiosk_service";
+import { useService } from "@web/core/utils/hooks";
+import { KioskTemplate } from "@pos_self_order/kiosk/template/kiosk_template";
+
+export class TableNumber extends Component {
+    static template = "pos_self_order.TableNumber";
+    static components = { KioskTemplate };
+
+    setup() {
+        this.selfOrder = useselfOrder();
+        this.selfOrder.isOrder();
+        this.router = useService("router");
+        this.state = useState({
+            tableNumber: "",
+        });
+    }
+
+    numberClick(event) {
+        const key = event.target.attributes.data.value;
+
+        if (key === "reset") {
+            this.state.tableNumber = this.state.tableNumber.slice(0, -1);
+        } else if (key === "clear") {
+            this.state.tableNumber = "";
+        } else {
+            this.state.tableNumber += key;
+        }
+    }
+
+    confirm() {
+        if (this.state.tableNumber.length > 0) {
+            this.selfOrder.tableNumber = this.state.tableNumber;
+            this.router.navigate("default");
+        }
+        this.router.navigate("payment");
```

Comment:
The behaviour here is strange. If the tableNumber length is > 0, then we navigate to default and then directly navigate to payment. If the tableNumber is not set, we navigate to payment. We should make this a bit clearer